taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -828,6 +709,21 @@ FunctionType OpenglCodeGen::gen(void) {
 #endif
 }
 
+void OpenglCodeGen::lower() {
+  auto ir = kernel_->ir;
+  auto &config = kernel_->program.config;
+  config.demote_dense_struct_fors = true;
+  auto res = irpass::compile_to_offloads(ir, config,
+                              /*vectorize=*/false, kernel_->grad,
+                              /*ad_use_stack=*/false, config.print_ir,
+                              /*lower_global_access*/true);
+  global_tmps_buffer_size_ = res.total_size;
+  TI_TRACE("[glsl] Global temporary buffer size {} B", global_tmps_buffer_size_);
+#ifdef _GLSL_DEBUG
+  irpass::print(ir);
+#endif
+}
```

Comment:
What about the passes after `offload`?

Reply:
What do you mean by `offload`? Can't get your point.. Sorry..

Reply:
Oh, I understand now. The return value of `compile_to_offloads` is obtained by `irpass::offload`, this doesn't mean that `ir` is also returned at that time... (I was wondering where are the passes after `irpass::offload` (such as `full_simplify` and `demote_atomics`).)